x/mongo/driver/topology/connection.go

```diff
@@ -226,9 +227,10 @@ func (c *connection) connect(ctx context.Context) {
 	handshakeConn := initConnection{c}
 	handshakeInfo, err = handshaker.GetHandshakeInformation(handshakeCtx, c.addr, handshakeConn)
 	if err == nil {
-		// We only need to retain the Description field as the connection's description. The authentication-related
+		// We only need to retain the Description and ServerConnectionID fields. The authentication-related
 		// fields in handshakeInfo are tracked by the handshaker if necessary.
 		c.desc = handshakeInfo.Description
+		c.serverConnectionID = handshakeInfo.ServerConnectionID
```

Comment:
It appears the base branch for this PR was created before https://github.com/mongodb/mongo-go-driver/pull/728 was merged, so it does not show the additional mutex lock/unlock that surround `c.desc = handshakeInfo.Description` in the `master` branch. Github doesn't report any merge conflicts, but I'm a little concerned about giving this a 👍 without seeing how this PR interacts with that change.

Please rebase on the latest `master` branch and make sure the `c.descMu.Lock()` and `c.descMu.Unlock()` only surround the `c.desc` assignment.

Reply:
Ah great, thanks for the heads up. Rebased and will wait for passing tests before merging.